src/backend/src/services/part-review.services.ts

```diff
@@ -0,0 +1,1154 @@
+
+    if (!review || review.deletedAt !== null) {
+      throw new NotFoundException('Part Review', reviewId);
+    }
+
+    const isAdminUser = await userHasPermission(creator.userId, organizationId, isAdmin);
+
+    if (review.userCreatedId !== creator.userId && !isAdminUser) {
+      throw new AccessDeniedAdminOnlyException('create part review popup');
+    }
+
+    const newPopup = await prisma.part_Review_Popup.create({
+      data: {
+        review: { connect: { partReviewId: reviewId } },
+        xCoord,
+        yCoord,
+        fileIndex,
+        title,
+        description: description ?? ''
+      },
+      ...getPartReviewQueryArgs
+    });
+    return newPopup;
+  }
+
+  /**
+   * Updates a part review popup
+   * @param organizationId id of the organization
+   * @param popupId ID of the popup to update
+   * @param xCoord New X coordinate
+   * @param yCoord New Y coordinate
+   * @param fileIndex the index of the file this popup is on
+   * @param title New title
+   * @param description New description
+   * @param updater The user updating the popup
+   * @returns The updated popup
+   */
+  static async updatePartReviewPopup(
+    organizationId: string,
+    popupId: string,
+    xCoord: number,
+    yCoord: number,
+    fileIndex: number,
+    title: string,
+    description: string,
+    updater: User
+  ) {
+    const popup = await prisma.part_Review_Popup.findUnique({ where: { partReviewPopupId: popupId } });
+
+    if (!popup || popup.deletedAt !== null) {
+      throw new NotFoundException('Pop Up', popupId);
+    }
+
+    const isAdminUser = await userHasPermission(updater.userId, organizationId, isAdmin);
+
+    if (!isAdminUser) {
+      throw new AccessDeniedAdminOnlyException('update part review popup');
+    }
+
+    return prisma.part_Review_Popup.update({
+      where: { partReviewPopupId: popupId },
+      data: { xCoord, yCoord, fileIndex, title, description, updatedAt: new Date() },
+      ...getPartReviewQueryArgs
+    });
+  }
+
+  /**
+   * Deletes a part review popup
+   * @param popupId ID of the popup to delete
+   * @param deleter The user deleting the popup
+   * @returns Confirmation message
+   */
+  static async deletePartReviewPopup(popupId: string, deleter: User, organizationId: string) {
+    const popup = await prisma.part_Review_Popup.findUnique({
+      where: { partReviewPopupId: popupId },
+      include: { review: { select: { userCreatedId: true, partReviewId: true } } }
+    });
+
+    if (!popup || popup.deletedAt) {
+      throw new NotFoundException('Pop Up', popupId);
+    }
+
+    const isAdminUser = await userHasPermission(deleter.userId, organizationId, isAdmin);
+
+    if (!isAdminUser) {
+      throw new AccessDeniedAdminOnlyException('delete part review popup');
+    }
+
+    await prisma.part_Review_Popup.update({
+      where: { partReviewPopupId: popupId },
+      data: { deletedAt: new Date() },
+      ...getPartReviewQueryArgs
+    });
```

Comment:
void return...

Reply:
weird that these last few functions dont follow the same pattern as the rest of the file is the only reason im making these comments, we should try to keep everything as consistent as possible